### ISSUE TITLE
Allow hiding documentation members that don't have descriptions

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot-proposals#8065]
### Describe the project you are working on

[A certain rather big addon.](https://github.com/KoBeWi/Metroidvania-System)

### Describe the problem or limitation you are having in your project

I have an advanced addon with its own documentation. However not all members in the documentation have description and it's by design. The members that don't have description are not supposed to be used by the user; they are internal. There is no way to hide them however. While in some cases it's a limitation of GDScript's docgen, some things can't be hidden at all (like signals). This unfortunately creates lots of noise and makes the class documentation less readable, because relevant members are mixed with irrelevant members.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add some new checkbox/option in the Editor Help. When enabled, the members that don't have description would be hidden entirely from the user.

Example:
Before
![image](https://github.com/godotengine/godot-proposals/assets/2223172/ebb92d85-9b2d-4bf3-878e-9afb275cfad4)
After
![image](https://github.com/godotengine/godot-proposals/assets/2223172/86b0a3c9-5ba9-4984-a20e-49b3963de8f6)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The option would be here probably.
![image](https://github.com/godotengine/godot-proposals/assets/2223172/abb98bbe-b177-4e26-8ae6-6ae5284bd5aa)
When toggled, the doc page would refresh and all things with no description would be skipped from being displayed.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No.

### Is there a reason why this should be core and not an add-on in the asset library?

It's a feature for making addons.

**Comment by Mickeon** [comment]
I do agree with the proposal, but the more suggestions arrive about "hiding things from the user", the more justifiable it feels to begin work on a `private` keyword, or some... `@hidden` annotation. It would be a catch-all solution of sorts.

**Comment by AThousandShips** [comment]
For example with doxygen this is often the default AFAIK, you just have the entry in the summary and no detailed entry

**Comment by YuriSizov** [comment]
I think the regular rules of culling everything prefixed with an underscore still apply here. If signals do not respect that, this should be addressed. But I think overall it should be as simple as that. Mark something private-ish by adding an underscore, no custom syntax.

**Comment by KoBeWi** [comment]
Only methods respect the underscore prefix (last time I tested at least).

**Comment by dalexeev** [comment]
I agree that the documentation should only skip undocumented private members (at the moment we can only recognize them by the underscore prefix). Public members should be displayed even if they do not have a description, because you may accidentally access them and not notice any differences from other public members.

If you don't want the user to use any API, then you should either make it private (preferred) or note in the documentation that the member is internal and should not be used.

> Only methods respect the underscore prefix (last time I tested at least).

I think we need to add a check for signals, enums, enum members, and classes.

<details>
<summary>Details</summary>

```gdscript
extends Node
var _var
const _CONST = 1
enum _Enum {A, B, C}
enum Enum {A, _B, C}
signal _signal()
func _func(): pass
class _Class: pass
```

![](https://github.com/godotengine/godot-proposals/assets/47700418/f43de20e-37ad-4404-8a9a-8066ed7e3361)
![](https://github.com/godotengine/godot-proposals/assets/47700418/d83d53a2-07c8-47c4-8d77-1f02825a31f5)

</details>

**Comment by OS-of-S** [comment]
I run into case where I want to exclude property from documentation but I can't use uderscore((((( It's because of prefixed group:

`@export_group("Marble Bag Advanced", "marble_bag")`

And then I have an editor button:

`@export_tool_button("Clear marble bag", "Clear") var marble_bag_editor_button: Callable = _clear_marble_bag_in_editor`

Prefix grouping removes prefix from displayed name and that's nore important for me, to make interface names nice and short... But it lead to appearance of this button in documentation... I just' don't know how to get rid of it.

<img width="721" height="110" alt="Image" src="https://github.com/user-attachments/assets/5bd4738a-ae09-4ff3-bce2-bec85cb8c39a" />
